esp/esp/program/controllers/lottery.py

```diff
@@ -697,6 +750,27 @@ def clear_saved_assignments(self, delete=False):
         else:
             old_registrations.filter(StudentRegistration.is_valid_qobject()).update(end_date=datetime.now())
 
+    def export_assignments(self):
+        def export_array(arr):
+            s = StringIO()
+            numpy.savetxt(s, arr)
+            return s.getvalue()
+
+        student_sections = export_array(self.student_sections)
+        student_ids = export_array(self.student_ids)
+        section_ids = export_array(self.section_ids)
+        return base64.b64encode(zlib.compress(student_sections + '|' + student_ids + '|' + section_ids))
+
+    def import_assignments(self, data):
+        data_parts = zlib.decompress(base64.b64decode(data)).split('|')
+
+        if len(data_parts) != 3:
+            raise ValueError('provided lottery_data is corrupted (doesn\'t contain three parts)')
+
+        self.student_sections = numpy.loadtxt(StringIO(data_parts[0]))
```

Comment:
How reasonable would it be to do this as JSON rather than `numpy`?  I'm kinda worried that `numpy.loadtxt` turns out to be hideously insecure, as e.g. `pickle` is.  (It doesn't super matter because this is admins, but if it's no harder it would be nice to be secure.)